src/main/java/sx/blah/discord/api/internal/DiscordClientImpl.java

```diff
@@ -178,8 +178,8 @@ private DiscordClientImpl(long timeoutTime, int maxMissedPingCount, boolean isDa
 		this.isBot = isBot;
 		this.reconnectAttempts = reconnectAttempts;
 		this.dispatcher = new EventDispatcher(this);
-		this.loader = new ModuleLoader(this);
 		this.shardCount = shardCount;
+		this.loader = new ModuleLoader(this);
```

Comment:
This Move is on purpose
The module loader constructor was throwing an NPE when it called isReady() and isReady() accessed shardCount